src/TableOfContents/index.tsx

```diff
@@ -0,0 +1,226 @@
+
+    const parentIndex = findParentIndex(contents[index].depth, contents.slice(0, index));
+    if (parentIndex > -1 && !expanded[parentIndex]) {
+      setExpanded({ ...expanded, [parentIndex]: true });
+    }
+  }, [srn]);
+
+  const isMobile = useIsMobile(enableDrawer);
+
+  const comp = (
+    <div className={cn('TableOfContents bg-gray-1 dark:bg-transparent flex justify-end h-full', className)}>
+      <div className="w-full">
+        <ScrollContainer>
+          <div className={cn('TableOfContents__inner ml-auto', `py-${padding}`)}>
+            {contents.map((item, index) => {
+              if (item.depth > 0) {
+                // Check if we should show this item
+                const parentIndex = findParentIndex(item.depth, contents.slice(0, index));
+                if (parentIndex > -1 && !expanded[parentIndex]) {
+                  return null;
+                }
+              }
+
+              const isGroup = item.type === 'group';
+              const isDivider = item.type === 'divider';
+              const isExpanded = expanded[index];
+
+              return (
+                <TableOfContentsItem
+                  key={index}
+                  name={item.name}
+                  // srn={item.srn}
+                  depth={item.depth}
+                  isActive={item.isActive || false}
+                  isGroup={isGroup}
+                  isExpanded={isExpanded}
+                  isDivider={isDivider}
+                  onClick={e => {
+                    if (isDivider) {
+                      e.preventDefault();
+                      return;
+                    }
+
+                    if (!isGroup) return;
+
+                    e.preventDefault();
+                    setExpanded({ ...expanded, [String(index)]: !isExpanded });
+                  }}
+                  icon={item.icon}
+                />
+              );
+            })}
+          </div>
+        </ScrollContainer>
+      </div>
+    </div>
+  );
+
+  if (isMobile) {
+    const { org, project } = deserializeSrn(srn || '');
+    return (
+      <Drawer isOpen={isOpen} onClose={() => onClose()} position="left" size="330px">
+        <div className="flex flex-1 flex-col bg-gray-1 dark:bg-transparent">
+          <div className="border-b dark:border-lighten-4 h-20 py-6 px-2 bg-white">
+            <Button className="flex justify-start text-lg" icon={'arrow-left'} minimal onClick={() => onClose()}>
+              {title || `${org} / ${project}`}
+            </Button>
+          </div>
+          <div className="h-full flex justify-end">{comp}</div>
+        </div>
+      </Drawer>
+    );
+  }
+
+  return comp;
+};
+
+interface ITableOfContentsItem {
+  depth: number;
+  name: string;
+  srn?: string;
+  isActive: boolean;
+  isGroup: boolean;
+  isExpanded: boolean;
+  isDivider: boolean;
+  onClick(e: React.MouseEvent<HTMLDivElement, MouseEvent>): void;
+  icon?: IconName;
+}
```

Comment:
Might be easier to just pass in the full item now that `IContentsNode` has more information on it?

```ts
interface ITableOfContentsItem {
  item: IContentsNode
  isExpanded: boolean;
  onClick(e: React.MouseEvent<HTMLDivElement, MouseEvent>): void;
}
```